tests/test_axis_limits.py

```diff
@@ -1,101 +1,324 @@
         eq = Equilibrium()
-        grid = LinearGrid(L=2, M=2, N=2, axis=False)
+        grid = LinearGrid(L=2, M=1, N=1, axis=False)
         assert not grid.axis.size
-        data = eq.compute("B0", grid=grid)
-        assert "B0" in data and "psi_r" in data and "sqrt(g)" in data
-        # assert axis limit dependencies are not in data
-        assert "psi_rr" not in data and "sqrt(g)_r" not in data
-        grid = LinearGrid(L=2, M=2, N=2, axis=True)
+        data = eq.compute(name, grid=grid).keys()
+        assert name in data and deps < data and axis_limit_deps.isdisjoint(data)
+        grid = LinearGrid(L=2, M=1, N=1, axis=True)
         assert grid.axis.size
-        data = eq.compute("B0", grid=grid)
-        assert "B0" in data and "psi_r" in data and "sqrt(g)" in data
-        # assert axis limit dependencies are in data
-        assert "psi_rr" in data and "sqrt(g)_r" in data
-        assert np.isfinite(data["B0"]).all()
+        data = eq.compute(name, grid=grid)
+        assert name in data and deps | axis_limit_deps < data.keys()
+        assert np.isfinite(data[name]).all()
 
     @pytest.mark.unit
-    def test_nonexistent_limits(self):
-        """Test that quantities whose limit does not exist evaluates not finite."""
-        eq = get("W7-X")
-        grid = LinearGrid(L=5, M=5, N=5, sym=eq.sym, NFP=eq.NFP, axis=True)
-        axis_mask = grid.nodes[:, 0] == 0
-        no_limits = ["e^theta", "grad(alpha)"]
-        data = eq.compute(names=no_limits, grid=grid)
-        for quantity in no_limits:
-            assert np.all(~np.isfinite(data[quantity][axis_mask]))
-
-    @staticmethod
-    def continuity(eq, name, expected_at_axis=None):
-        """Test that the rho=0 axis limit of name is computed accurately."""
-        delta = 1e-5
-        epsilon = 1e-5
-        rho = np.linspace(0, 1, 10) * delta
-        grid = LinearGrid(rho=rho, M=7, N=7, NFP=eq.NFP, sym=eq.sym)
-        assert grid.axis.size
-        quantity = eq.compute(name, grid=grid)[name]
-        if (
-            data_index["desc.equilibrium.equilibrium.Equilibrium"][name]["coordinates"]
-            == "r"
-        ):
-            quantity = compress(grid, quantity)
-        elif (
-            data_index["desc.equilibrium.equilibrium.Equilibrium"][name]["coordinates"]
-            != ""
-        ):
-            quantity = surface_averages(grid, quantity, expand_out=False)
-        assert np.isfinite(quantity).all()
-        # check continuity
-        np.testing.assert_allclose(quantity[:-1], quantity[1:], atol=epsilon)
-
-        # check expected value at axis
-        if expected_at_axis is None:
-            # fit the data (except axis pt) to a polynomial to extrapolate to axis
-            poly = np.polyfit(rho[1:], quantity[1:], 6)
-            expected_at_axis = poly[-1]  # constant term is same as eval poly at rho=0
-        np.testing.assert_allclose(quantity[0], expected_at_axis, atol=epsilon)
+    def test_limit_existence(self):
```

Comment:
You added this new test but it looks like it is not actually used?

Reply:
It is not used since it won't pass. All the limits are implemented in #556.
I pulled this file from #556 so that reviewers know that the `test_data_index_deps` test passes. (Manually checking that there are no extra dependencies in the compute functions for the basis vectors would be a chore).